apps/src/templates/rubrics/LearningGoal.jsx

```diff
@@ -38,6 +57,78 @@ export default function LearningGoal({
     setIsOpen(!isOpen);
   };
 
+  const handleFeedbackChange = event => {
+    if (studentLevelInfo.user_id && learningGoal.id) {
+      if (autosaveTimer.current) {
+        clearTimeout(autosaveTimer.current);
+      }
+      teacherFeedback.current = event.target.value;
+      setDisplayFeedback(teacherFeedback.current);
+      autosaveTimer.current = setTimeout(() => {
+        autosave();
+      }, saveAfter);
+    }
+  };
+
+  const autosave = () => {
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoal.id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    HttpClient.put(`${base_endpoint}/${learningGoalEval.id}`, bodyData, true, {
+      'Content-Type': 'application/json',
+    })
+      .then(() => {
+        setIsAutosaving(false);
+        setAutosaved(true);
+      })
+      .catch(error => {
+        console.log(error);
+        setIsAutosaving(false);
+        setErrorAutosaving(true);
+      });
+    clearTimeout(autosaveTimer.current);
+  };
+
+  useEffect(() => {
+    if (studentLevelInfo && learningGoal.id) {
```

Comment:
I think you also want `canProvideFeedback` to be a condition here.

One thing I'm struggling with is that this component is also used in the student view and will need to fetch evaluations for that student. I think these can be in the same component for now (we may refactor it later) but I want to be careful about when we're issuing these requests

Reply:
That makes sense. Will probably want a different version of this for the student view that won't pull all of the data.

Reply:
By "this" do you mean this component or this useEffect?

Curious what the student view looks like with this change though. If it's not broken, I don't think it blocks this PR.

Reply:
Yep that sounds good!